arbor/include/arbor/util/variant.hpp

```diff
@@ -584,16 +614,16 @@ namespace std {
 // Unambitious hash:
 template <typename... T>
 struct hash<::arb::util::variant<T...>> {
-    std::size_t operator()(const ::arb::util::variant<T...>& v) {
+    std::size_t operator()(const ::arb::util::variant<T...>& v) const {
         return v.index() ^
-            visit([](const auto& a) { return std::hash<std::remove_cv_t<decltype(a)>>{}(a); }, v);
+            ::arb::util::visit([](const auto& a) { return std::hash<std::remove_cv_t<std::remove_reference_t<decltype(a)>>>{}(a); }, v);
```

Comment:
Can you replace `remove_cv<remove_reference<>>` with `decay<>`?

Reply:
It's not _quite_ the same (it doesn't turn arrays into pointers), so I think it matches the semantics more precisely.